lib/solargraph/source_map/mapper.rb

```diff
@@ -147,6 +147,23 @@ def process_directive source_position, comment_position, directive
               pins.last.docstring.add_tag YARD::Tags::Tag.new(:param, '', pins.last.return_type.to_s.split(', '), 'value')
             end
           end
+        when 'visibility'
+          begin
+            kind = directive.tag.text&.to_sym
+            return unless kind == :private || kind == :protected
+
+            name = directive.tag.name
+            namespace = closure_at(source_position) || @pins.first
+            if namespace.location.range.start.line < comment_position.line
+              namespace = closure_at(comment_position)
+            end
+
+            matches = pins.select{ |pin| pin.is_a?(Pin::Method) && pin.name == name && pin.namespace == namespace && pin.context.scope == namespace.is_a?(Pin::Singleton) ? :class : :instance }
+            matches.each do |pin|
+              # @todo Smelly instance variable access
```

Comment:
is there a reason for not having setters for the different variables?

Reply:
This is a wart left over from the rules that control how pins can be updated in place. Pin attributes are typically not writable because there are finite cases where modifying a pin won't adversely affect the ApiMap. Those instance variable accesses will eventually get refactored.